pkg/migrator/cleanup.go

```diff
@@ -0,0 +1,246 @@
+		return err
+	}
+	for _, csc := range cscs.Items {
+		err = kubeClient.Delete(context.TODO(), &csc)
+		if err != nil {
+			logger.Errorf("[migration] Failed to delete CatalogSourceConfig %s. Error: %s", csc.GetName(), err.Error())
+		} else {
+			logger.Printf("[migration] Succesfully deleted datastore CSC %s", csc.GetName())
+		}
+	}
+	return nil
+}
+
+// findCatalogSource infers the datastore CatalogSource that a given existing Subscription will
+// contain information about in it's spec.source and spec.sourceNamespace fields
+func findCatalogSource(subscription *olm.Subscription, client client.Client) (*olm.CatalogSource, error) {
+	associatedCscName := subscription.GetLabels()[builders.OwnerNameLabel]
+	possibleCsName, err := extractCsName(associatedCscName)
+	possibleCsNamespace := subscription.GetLabels()[builders.OwnerNamespaceLabel]
+	if err != nil {
+		return nil, err
+	}
+	// try and fetch the CatalogSource
+	datastoreCs := &olm.CatalogSource{}
+	namespacedName := types.NamespacedName{Name: possibleCsName, Namespace: possibleCsNamespace}
+	err = client.Get(context.TODO(), namespacedName, datastoreCs)
+	if err != nil {
+		return nil, err
+	}
+	return datastoreCs, nil
+}
+
+// findCsFromOpsrc extracts the packageName from a Subscription,
+// and finds the corresponding CatalogSource that the package belongs to.
+func findCsFromOpsrc(subscription *olm.Subscription, kubeClient client.Client) (*olm.CatalogSource, error) {
+	packageName := getSubsPackageName(subscription)
+	opsrcs := &v1.OperatorSourceList{}
+	err := kubeClient.List(context.TODO(), &client.ListOptions{}, opsrcs)
+	if err != nil {
+		return nil, err
+	}
+	for _, instance := range opsrcs.Items {
+		if !isPackageInOpsrc(packageName, &instance) {
+			continue
+		}
+		// fetch the CatalogSource with the same name
+		datastoreCs := &olm.CatalogSource{}
+		namespacedName := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
+		err = kubeClient.Get(context.TODO(), namespacedName, datastoreCs)
+		if err != nil {
+			return nil, err
+		}
+		return datastoreCs, nil
+	}
+	return nil, nil
+}
+
+// isPackageInOpsrc takes the name of a package, and an OperatorSource
+// as input, and returns true if the package is present in the OperatorSource's
+// status.Packages field
+func isPackageInOpsrc(packageName string, opsrc *v1.OperatorSource) bool {
+	packages := getOpsrcPackages(opsrc)
+	for _, pkg := range packages {
+		if pkg == packageName {
+			return true
+		}
+	}
+	return false
+}
+
+// extractCsName takes the name of a CatalogSourceConfig
+// as input, and infers the name of the corresponding
+// CatalogSource datastore. Installed CSCs follow the
+// following naming pattern: installed-publisher-namespace.
+// For example, for a CatalogSourceConfig named
+// `installed-community-openshift-operators`,
+// extractCsName returns `community-operators` as output
+func extractCsName(cscName string) (string, error) {
+	var stringBuilder strings.Builder
```

Comment:
strings.Builder seems like slight overkill here, no? If we are just appending two strings why not just use `fmt.Sprintf("%s-%s", possibleCsName, "-operators")`?